src/App.tsx

```diff
@@ -1,98 +1,62 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Autocomplate } from './component/Autocomplete';
+import { debounce } from './utils/debounce';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isInputFocused, setIsInputFocused] = useState(false);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
+
+  const currentPerson = peopleFromServer.find(
+    person => person.name === selectedPerson,
+  );
+
+  const handleQueryClick = (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
+    setQuery(event.currentTarget.textContent || '');
+    setSelectedPerson(event.currentTarget.textContent || '');
+    setIsInputFocused(true);
```

Comment:
```suggestion
    event.preventDefault();
    setQuery(event.currentTarget.textContent || '');
    setSelectedPerson(event.currentTarget.textContent || '');
```
no need to manage focus here. it should make onFocus and onBlur handlers.
better to pass person in function and set work with person or person.name instead event.currentTarget.textContent.